content/articles/securing-mongodb-in-linux/index.md

```diff
@@ -0,0 +1,230 @@
+
+```bash
+$mongo
+```
+
+The output displays the shell's version, the connection of MongoDB that is its IP address, and more warnings indicating database access control is not enabled. This shows that reading and writing to the database is unrestricted, insecure and anyone with access to the server can have control of the database. Due to this, we can check the list of databases available to the server. By running the command below:
+
+```bash
+#The command returns lists of every database on the server.
+show dbs
+```
+
+ something like this will appear:
+
+```output
+db-job-admin   0.0091GB
+db-jobs-new  0.0009GB
+db-config  0.0001GB
+local   0.0100GB
+```
+
+The output will display default databases, and any user can still find important data in one system. To end this situation, we add an administrative user by connecting to the database. To start connecting to the database, we call an admin user to the database using the command below:
+
+```bash
+use admin
+```
+
+Since the MongoDB environment comes installed with several JSON-based methods used in managing the database, we will use the javascript-based shell method that is used to create a user. The `db.createUser` method is used to add new users to the database. To run the db.createUser method, we use the command:
+
+```bash
+db.createUser({user, getLastError})
+#field user contains the documents with authentication information about the user.
+#getlasError can be optional, writes and checks if the user exists.
+```
+
+The method requires users to create a new user, if exists, an error message will return showing that the database already exists. when creating the user, you need to specify login credentials, passwords, and roles of the user to the database. As we have stated before, MongoDB stores data in JSON documents that hold data in its fields. For a JSON document to execute without any error message, one need to use curly braces to start and end the document created as shown below:
+
+```bash
+#note that without closing parenthesis MongoDB will still not run.
+    use admin
+    db.createUser({
+        user: "AndrewKulundeng'",
+        pwd: "xandrew12@12",#enter roles, closing braces and parenthis.
+        role: [{role:"userAdminAnyDatabase", dp:"admin"}]})
+```
+
+The method executes properly to prompt the user to input password as shown:
+
+```Output
+*enter password used when creating the database, press enters to execute.*
+Enter password: xandrew12@12
+Output
+Successfully added user: {
+     user : "AndrewKulundeng'",
+     roles : [
+        {
+            "role" : "userAdminAnyDatabase",
+            "db" : "admin"
+        }
+      
+    ]
+}
+
+```
+
+Exist MongoDB after the operation. Users will be allowed to enter details after authentication is enabled and the MongoDB daemon restarts. we can as well restart using the command below:

```

Comment:
What does, "Exist MongoDB after the operation" mean?

Additionally, on the third sentence, you cannot start a sentence with a small letter.